packages/native-stack/src/utils/prepareHeaderBarButtonItems.ts

```diff
@@ -0,0 +1,82 @@
+import { Image, processColor } from 'react-native';
+
+import type {
+  NativeStackHeaderBarButtonItem,
+  NativeStackHeaderBarButtonItemWithCustomView,
+  NativeStackHeaderBarButtonItemWithMenu,
+} from '../types';
+
+const prepareMenu = (
+  menu: NativeStackHeaderBarButtonItemWithMenu['menu'],
+  index: number,
+  routeKey: string,
+  side: 'left' | 'right'
+): any => {
+  return {
+    ...menu,
+    items: menu.items.map((menuItem, menuIndex) => {
+      if ('items' in menuItem) {
+        return prepareMenu(menuItem, menuIndex, routeKey, side);
+      }
+      return {
+        ...menuItem,
+        menuId: `${menuIndex}-${index}-${routeKey}-${side}`,
+      };
+    }),
+  };
+};
+
+export const prepareHeaderBarButtonItems = (
+  barButtonItems: (
+    | NativeStackHeaderBarButtonItem
+    | NativeStackHeaderBarButtonItemWithCustomView<any>
```

Comment:
Don't use `any`. If it can be anything, use `unknown` instead.

It'll be better to add a generic to the function and use it for the proper type.